src/shared/varlink.c

```diff
@@ -1219,17 +1220,23 @@ int varlink_close(Varlink *v) {
 Varlink* varlink_close_unref(Varlink *v) {
         if (!v)
                 return NULL;
-
-        (void) varlink_close(v);
+                                   /* n_ref >= 1 */
+        varlink_ref(v);            /* n_ref >= 2 */
+        varlink_close(v);          /* n_ref >= 1 */
+        if (v->n_ref > 1)
+                v->n_ref--;        /* n_ref >= 1 */
         return varlink_unref(v);
 }
 
 Varlink* varlink_flush_close_unref(Varlink *v) {
         if (!v)
                 return NULL;
-
-        (void) varlink_flush(v);
-        (void) varlink_close(v);
+                                   /* n_ref >= 1 */
+        varlink_ref(v);            /* n_ref >= 2 */
+        varlink_flush(v);          /* n_ref >= 1 */
```

Comment:
`varlink_flush()` does not decrease the reference counter. So the `/* n_ref >= 1 */` comment should moved one line below.

Maybe, more simply, it is better the following:
```
(void) varlink_flush(v);
return varlink_close_unref(v);
```

Reply:
It *currently* doesn't. I included `varlink_flush` under protection for that reason. But yeah, maybe that's too much. I'll update code as suggested.